### ISSUE TITLE
Assertion and crash when loading saved mission at Terranova 4 (9.4)

### ISSUE THREAD

**Issue by Anton-V-K** [colobot/colobot#933]
I've saved the mission as usual (no programs were running on bots) in 0.1.9 (Windows 7).
And I'm unable to load it back: assertion is followed by a crash if I press `Ignore':
```
Program: C:\Program Files\Colobot Gold Edition\colobot.exe
File: C:/Users/krzys/Desktop/colobot/src/object/implementation/programmable_impl.cpp, Line 158

Expression: op < static_cast<int>(dynamic_cast<CProgramStorageObject*>(m_object)->GetProgramCount())
```
The saved mission is attached as [saveTerra4.zip](https://github.com/colobot/colobot/files/836939/saveTerra4.zip) and the log is in [log.txt.zip](https://github.com/colobot/colobot/files/836937/log.txt.zip).
The problem happens with all saved files at this mission (even when auto-saved).

Until now I had no problems with loading saved missions at other levels.

I see that similar issue #765 had already been fixed (though it missed the official release), so probably it is worth checking whether I've encountered same problem.

**Comment by melex750** [comment]
Thanks for reporting. This is the same bug as #765. Fixed in 1b074bd dev.

Four AlienAnts were improperly saved. [saveTerra4fixed.zip](https://github.com/colobot/colobot/files/843239/saveTerra4fixed.zip)

**Comment by RaptorParkowsky** [comment]
If it doesn't happen on latest dev, then it's basically duplicate.
